wgpu-native/src/device.rs

```diff
@@ -1182,19 +1184,33 @@ pub fn device_create_bind_group<B: GfxBackend>(
         for (b, decl) in bindings.iter().zip(&bind_group_layout.bindings) {
             let descriptor = match b.resource {
                 binding_model::BindingResource::Buffer(ref bb) => {
+                    let buffer_usage = buffer_guard[bb.buffer].usage;
                     let (alignment, usage) = match decl.ty {
                         binding_model::BindingType::UniformBuffer => {
+                            assert!(
```

Comment:
we don't need to do the checks right in these arms: we get `usage` as a result of the match, and `buffer` a few lines afterwards, so we can just do a single check at that point